Added initial implementation of loader - spinner

nitinja opened the pull request.

Commit:
Added initial implementation of loader - spinner

nitinja commented:
I am working on eleventy page demo & react demo for this component

nitinja commented:
Eleventy page showing preview correctly in dev mode, but not on deployed preview 
![image](https://user-images.githubusercontent.com/3931042/126633498-a463cabd-66f9-4385-a2f8-ca45136d1725.png)

sandykadam commented:
@khawkins98 @nitinja just curious to know if we can build our own EMBL style loader/spinner instead of standard one.

Like we can use the hexagon icon of logo and circle around it animated effect.

https://monophy.com/gifs/embl-embllogo-embllogomonochrome-SU8RkgEO4bcb9bL1U9

https://www.embl.org/news/wp-content/uploads/2014/09/embl_virus_logo_animation_5sec.gif

nitinja commented:
![Animation](https://user-images.githubusercontent.com/3931042/126636479-7a328e14-060f-4c4d-8a37-720326b5f799.gif)

sandykadam commented:
Ah looks nice 👍

khawkins98 commented on components/vf-indicator-loader/README.md:
```diff
@@ -0,0 +1,46 @@
+
+## About
+
+An anchor or JavaScript button to scroll the  user to top of the current page, or to a target element.
```
I think this is old copy?

khawkins98 committed:
Update package.json

Lerna will bump this to alpha.1 on release

khawkins98 commented on components/vf-indicator-loader/vf-indicator-loader.njk:
```diff
@@ -0,0 +1,18 @@
+
+        <div class="vf-indicator-loader--spinner__circle"></div>
+
+{% if example == true %}
```
further above you used ` {% if container == true %}`... did you mean to `container` here as well?

khawkins98 commented on components/vf-indicator-loader/vf-indicator-loader.scss:
```diff
@@ -0,0 +1,51 @@
+.vf-indicator-loader--spinner__circle {
+  width: 32px;
+  height: 32px;
+  color: $vf-indicator-loader--border-color;
```
I have some suggestions on how to use the design tokens here. Might be clearest/best if I just do that and then we can look at the diff.

khawkins98 replied:
@nitinja I still need to update this Sass, but if you could please look at the other PR comment.

khawkins98 replied:
I've addressed this with commit https://github.com/visual-framework/vf-core/pull/1631/commits/0c370a6d600bd834fd3f090b0a8a323f9d347170

khawkins98 commented:
Looking good @nitinja — I did have a few minor things that I've noted.

sandykadam commented:
> Looking good @nitinja — I did have a few minor things that I've noted.

Are we good to use hexagon style or grey circle loader

nitinja commented:
@sandykadam I have committed just the circle spinner as of now, I think we can't commit any design that's not approved by UX people.

khawkins98 committed:
Use CSS custom props

This does a couple of things:

1. Sass linting to alphabitise ordering
2. use CSS vars (custom props) — we tend to use this over the Sass vars as this allows more possibility to cascade in new values for styling

khawkins98 committed:
Readme tweaks

khawkins98 commented:
Adding a note here that based on our recent conversation with @cindyebi and @nitinja this will be part of `vf-progress-indicator`, and specifically a `vf-progress-indicator--spinner` variant.
